src/test/java/gregtech/api/capability/impl/MultiblockRecipeLogicTest.java

```diff
@@ -546,4 +550,176 @@ protected List<IItemHandlerModifiable> getInputBuses() {
+            @Override
+            public boolean isAttachedToMultiBlock() {
+                return true;
+            }
+
+            @Override
+            public MultiblockControllerBase getController() {
+                return mbt;
+            }
+        };
+        MetaTileEntityFluidHatch importFluidBus2 = new MetaTileEntityFluidHatch(gregtechId("fluid_hatch.import.lv"), 1, false) {
+            @Override
+            public boolean isAttachedToMultiBlock() {
+                return true;
+            }
+
+            @Override
+            public MultiblockControllerBase getController() {
+                return mbt;
+            }
+        };
+        MetaTileEntityFluidHatch exportFluidBus = new MetaTileEntityFluidHatch(gregtechId("fluid_hatch.export.lv"), 1, true) {
+            @Override
+            public boolean isAttachedToMultiBlock() {
+                return true;
+            }
+
+            @Override
+            public MultiblockControllerBase getController() {
+                return mbt;
+            }
+        };
+
+        //Controller is a private field but we need that information
+        try {
+            Field field = MetaTileEntityMultiblockPart.class.getDeclaredField("controllerTile");
+            field.setAccessible(true);
+            field.set(importItemBus, mbt);
+            field.set(importFluidBus, mbt);
+            field.set(importFluidBus2, mbt);
+            field.set(exportFluidBus, mbt);
+        } catch (NoSuchFieldException | IllegalAccessException e) {
+            e.printStackTrace();
+        }
+
+        MultiblockRecipeLogic mbl = new MultiblockRecipeLogic(mbt) {
+
+            @Override
+            protected long getEnergyStored() {
+                return Long.MAX_VALUE;
+            }
+
+            @Override
+            protected long getEnergyCapacity() {
+                return Long.MAX_VALUE;
+            }
+
+            @Override
+            protected boolean drawEnergy(int recipeEUt, boolean simulate) {
+                return true;
+            }
+
+            @Override
+            public long getMaxVoltage() {
+                return 32;
+            }
+
+            // since the hatches were not really added to a valid multiblock structure,
+            // refer to their inventories directly
+            @Override
+            protected IItemHandlerModifiable getInputInventory() {
+                return importItemBus.getImportItems();
+            }
+
+            @Override
+            protected IMultipleTankHandler getInputTank() {
+                return new FluidTankList(false, importFluidBus.getImportFluids().getTankAt(0), importFluidBus2.getImportFluids().getTankAt(0));
+            }
+
+            @Override
+            protected IMultipleTankHandler getOutputTank() {
+                return exportFluidBus.getExportFluids();
+            }
+        };
+
```

Comment:
missing adding the item to the input bus, with the same comments about the OreDictUnifier returning an empty itemstack